xlators/storage/posix/src/posix-gfid-path.c

```diff
@@ -117,7 +118,19 @@ posix_get_gfid2path(xlator_t *this, inode_t *inode, const char *real_path,
             if (size == 0)
                 goto done;
         }
-        list = alloca(size);
+
+        /* iot_worker stack size is 256k, in case of a file having huge number
+           of backend xattrs size could be greater than 256k and iot_worker
+           thread is crashed so to avoid a crash use MALLOC if size is greater
+           than 32k
+        */
+        if (size > 32768) {
```

Comment:
I would say that 32 KiB is still a too big thing to have in a 256 KiB stack. I would always use `GF_MALLOC()` or limit it to 4 KiB at most (or even less).

Reply:
Agree - I'm not sure we are gaining anything by alloca() or the more complex code path here.

Reply:
I am also not sure about it, i have to generate the data.

Reply:
i have executed some basic test case (smallfile), i have not found any regression so i believe we can replace alloca with malloc.